### PR TITLE
add two udfs

### PR DESCRIPTION
**Pull request by XuetaoYin**

### CODE REVIEW

File: src/main/java/com/newsbreak/data/udf/ComputeLabelUDF.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+@Description(name = "ComputeLabelUDF", value = "ComputeLabelUDF(checked, clicked, shared, pv_time, cv_time, vv_time, thumb_up, thumb_down)",
+        extended = "")
+public class ComputeLabelUDF extends GenericUDF {
+    private ObjectInspectorConverters.Converter[] converters;
+
+    @Override
+    public ObjectInspector initialize(ObjectInspector[] arguments) throws UDFArgumentException {
+        if (arguments.length != 8) {
+            throw new UDFArgumentLengthException(
+                    "ComputeLabelUDF(checked, clicked, autoplay, shared, liked, thumbed_up, thumbed_down, pv_time, cv_time, vv_time)");
```
Comment (yi-duan-nb) [inline src/main/java/com/newsbreak/data/udf/ComputeLabelUDF.java:29]: nit: missing ctype, progress

File: src/main/java/com/newsbreak/data/udf/ComputeLabelUDF.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+import org.apache.hadoop.io.Text;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/**
+ * ComputeLabelUDF
+ * @author Xuetao
+ */
+
+@Description(name = "ComputeLabelUDF", value = "ComputeLabelUDF(checked, clicked, shared, pv_time, cv_time, vv_time, thumb_up, thumb_down)",
```
Comment (yi-duan-nb) [inline src/main/java/com/newsbreak/data/udf/ComputeLabelUDF.java:20]: nit: missing ctype, progress

File: src/main/java/com/newsbreak/data/udf/ComputeLabelUDF.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        extended = "")
+public class ComputeLabelUDF extends GenericUDF {
+    private ObjectInspectorConverters.Converter[] converters;
+
+    @Override
+    public ObjectInspector initialize(ObjectInspector[] arguments) throws UDFArgumentException {
+        if (arguments.length != 8) {
```
Comment (yi-duan-nb) [inline src/main/java/com/newsbreak/data/udf/ComputeLabelUDF.java:27]: length should be 10?

File: src/main/java/com/newsbreak/data/udf/ComputeLabelUDF.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        converters[10] = ObjectInspectorConverters.getConverter(arguments[10],
+                PrimitiveObjectInspectorFactory.writableStringObjectInspector);
+        converters[11] = ObjectInspectorConverters.getConverter(arguments[11],
+                PrimitiveObjectInspectorFactory.writableFloatObjectInspector);
+        return PrimitiveObjectInspectorFactory.javaStringObjectInspector;
+    }
+
+    @Override
+    public Object evaluate(DeferredObject[] arguments) throws HiveException, JSONException {
+        assert (arguments.length == 10);
```
Comment (yi-duan-nb) [inline src/main/java/com/newsbreak/data/udf/ComputeLabelUDF.java:55]: 12

File: src/main/java/com/newsbreak/data/udf/ComputeLabelUDF.java [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+        }
+
+        short checked = (Short) converters[0].convert(arguments[0].get());
+        short clicked = (Short) converters[1].convert(arguments[1].get());
+        short autoplay = (Short) converters[2].convert(arguments[2].get());
+        short shared = (Short) converters[3].convert(arguments[3].get());
+        short liked = (Short) converters[4].convert(arguments[4].get());
+        short thumbed_up = (Short) converters[5].convert(arguments[5].get());
+        short thumbed_down = (Short) converters[6].convert(arguments[6].get());
+        int pv_time = (Integer) converters[7].convert(arguments[7].get());
```
Comment (yi-duan-nb) [inline src/main/java/com/newsbreak/data/udf/ComputeLabelUDF.java:68]: i find many pv_time/cv_time/vv_time is null. will this cause NPE here?

File: src/main/java/com/newsbreak/data/udf/ComputeLabelUDF.java [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+        int vv_time = (Integer) converters[9].convert(arguments[9].get());
+        String ctype = (String) converters[10].convert(arguments[10].get());
+        float progress = (Float) converters[11].convert(arguments[11].get());
+
+        return AssignLabel(ctype, checked, clicked, autoplay, shared, liked, thumbed_up, thumbed_down, pv_time, cv_time, vv_time, progress);
+    }
+
+    @Override
+    public String getDisplayString(String[] children) {
+        assert (children.length == 10);
```
Comment (yi-duan-nb) [inline src/main/java/com/newsbreak/data/udf/ComputeLabelUDF.java:79]: 12

File: src/main/java/com/newsbreak/data/udf/ComputeLabelUDF.java (hunk 1 above)
Comment (XuetaoYin) [reply src/main/java/com/newsbreak/data/udf/ComputeLabelUDF.java]: added

File: src/main/java/com/newsbreak/data/udf/ComputeLabelUDF.java (hunk 3 above)
Comment (XuetaoYin) [reply src/main/java/com/newsbreak/data/udf/ComputeLabelUDF.java]: yes updated

File: src/main/java/com/newsbreak/data/udf/ComputeLabelUDF.java (hunk 2 above)
Comment (XuetaoYin) [reply src/main/java/com/newsbreak/data/udf/ComputeLabelUDF.java]: added.

File: src/main/java/com/newsbreak/data/udf/ComputeLabelUDF.java (hunk 4 above)
Comment (XuetaoYin) [reply src/main/java/com/newsbreak/data/udf/ComputeLabelUDF.java]: fixed.

File: src/main/java/com/newsbreak/data/udf/ComputeLabelUDF.java (hunk 6 above)
Comment (XuetaoYin) [reply src/main/java/com/newsbreak/data/udf/ComputeLabelUDF.java]: fixed.

File: src/main/java/com/newsbreak/data/udf/ComputeLabelUDF.java (hunk 5 above)
Comment (XuetaoYin) [reply src/main/java/com/newsbreak/data/udf/ComputeLabelUDF.java]: Thanks.

**Review by yi-duan-nb — APPROVED** [review]

**Pull request merged**